src/nvim/event/wstream.c

```diff
@@ -73,6 +73,26 @@ bool wstream_write(Stream *stream, WBuffer *buffer)
   // This should not be called after a stream was freed
   assert(!stream->closed);
 
+  uv_buf_t uvbuf;
+  uvbuf.base = buffer->data;
+  uvbuf.len = UV_BUF_LEN(buffer->size);
+
+  if (!stream->uvstream) {
```

Comment:
i'm trying to understand, do we indicate anywhere that the lack of `stream.uvstream` implies a `stream.fd` ?  Should this be documented on `struct stream` , or a brief comment here? Normally this decided by a discriminator field...

In other places, the lack of `stream.uvstream` is taken to imply the presence of `stream.uv.idle`. 

https://github.com/neovim/neovim/blob/f3b7444e6638095305d86fa96fe0b7407c9b1648/src/nvim/event/rstream.c#L47-L51

Reply:
If you look at `fread_idle_cb`, you'll see that it reads from `stream->s.fd`.

Reply:
good point, but this is all very subtle, no? A brief docstring would really help (and an assert couldn't hurt). 

I see your latest push, thanks!